packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -160,11 +178,21 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
         case 'loading':
         case 'loadingMore':
         case 'sorting':
+        case 'filtering':
+          // If there isn't a abortController provided by the action and it is a filtering action (aka clientside filtering), it is an filterText update
```

Comment:
Why is this a separate case from the below? And why would the `abortController` not be set? I think we should probably abort the previous filtering request when the text updates.

Reply:
This is a separate case in consideration of the two kinds of "filtering" actions that can take place aka client side filtering vs server side filtering.

For client side filtering, I do a `dispatch` instead of a `dispatchFetch` and I don't include an `abortController` since client side filtering should just change the filterText in `data`. Its basically just an "update" action and thus doesn't interrupt any fetches that are currently occurring.

For server side filtering, it has an `abortController` and properly aborts previous filtering requests, acting similarly to the other actions.

I guess I could change it to be a `update` action for client side filtering, but does the above logic make sense? Lemme know if there are any flaws in that approach, greatly appreciate a second opinion